aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -232,6 +235,31 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
+template<typename T1, typename T2, typename Operation>
+void cummax_cummin_helper(const T1* self_data, T1* values_data, T2* indices_data,
+          int self_dim_size, int self_stride, int values_stride, int indices_stride) {
+      Operation op;
+      T1 out = self_data[0];
+      int idx = 0;
+      for(int i = 0; i < self_dim_size; i++) {
+        T1 curr_elem = self_data[i*self_stride];
+        if(std::isnan(curr_elem) || (!std::isnan(out) && op(curr_elem, out))) {
```

Comment:
Well, in MSVC, calling `std::isnan` on an integral object will confuse the compiler because `std::isnan<IntegralType>` is not implemented. This is causing the Windows build to fail.

Reply:
I proposed a fix in https://github.com/pytorch/pytorch/pull/33483. Would you please take a look there?